hdt-lib/src/hdt/BasicHDT.cpp

```diff
@@ -656,7 +657,7 @@ void BasicHDT::loadFromHDT(std::istream & input, ProgressListener *listener)
 	controlInformation.load(input);
 	std::string hdtFormat = controlInformation.getFormat();
 	if(hdtFormat!=HDTVocabulary::HDT_CONTAINER) {
```

Comment:
Jup indeed, I sort of started on a parse function: https://github.com/rdfhdt/hdt-cpp/blob/61288cbfb1d0fb3424392d26ea6d20c77d5b8024/hdt-lib/include/HDTVersion.hpp#L9